corehq/ex-submodules/couchforms/const.py

```diff
@@ -18,17 +22,47 @@
                   TAG_NAME, TAG_META, ATTACHMENT_NAME, 'case', MAGIC_PROPERTY]
 
 
-class BadRequest(object):
+# Should be in sync with file extensions supported by CommCare
+# Copied from https://github.com/dimagi/commcare-android/blob/62ae0bde9ed623db17bcc44ad678d693e0b73cb6/app/src/org/commcare/utils/FormUploadUtil.java#L57
+SUPPORTED_MEDIA_FILE_EXTENSIONS = [
+    "jpg", "jpeg", "3gpp", "3gp", "3ga", "3g2", "mp3", "wav", "amr",
+    "mp4", "3gp2", "mpg4", "mpeg4", "m4v", "mpg", "mpeg", "qcp", "ogg"
+]
+PERMITTED_FORM_SUBMISSION_FILE_EXTENSIONS = ['xml']
+
 
+class InvalidRequest(ABC):
```

Comment:
Don't you think these should be exceptions that get raised instead of having to do an `isinstance` check?

Reply:
Do you mean raising an exception and the error message be the "message" we are putting on the instance?
if yes, there is actually a need to also support a different http code. I think i originally used the exception route but then in order to support http code as well, I chose to have them as separate classes to have a clean implementation.
Let me know if you think that this can still be done via exceptions itself and support http code too in a way better than having classes.

Reply:
you can have separate exception classes and still have the `response` method to build the response with different codes.